android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -90,6 +105,25 @@ class RiveReactNativeView(private val context: ThemedReactContext) : FrameLayout
     addView(riveAnimationView)
   }
 
+  override fun onAttachedToWindow() {
+    // RiveReactNativeView may attach multiple times for any reason, so using
+    // a flag here isAttached to determine when it was initially attached to
+    // only addView once
+    if (childCount == 0 && isAttached == false) {
```

Comment:
What conditions cause it to be attached multiple times?

Reply:
Specifically with our Example App, it uses this navigation lib `@react-navigation/stack` which seems to be responsible for the attach/detach multiple times behavior. 
```
// On selecting an example in our example app
attach -> detach -> attach

// On hitting the back button to go back to the examples homepage
detach -> attach -> detach
```

But this could be the case for a different scenario created by a consumer too, so this is kind of a failsafe to try and coordinate the attach/detach logic of `RiveReactNativeView` here with `RiveAnimationView` of Android.

Tracking the `riveAnimationView.isAttachedToWindow` does not seem to be reliable here, so tracking the `childCount` to determine whether to addView seems to do the trick here, but open to other alternatives to try out!

With detaching in particular, this is the problematic bit because detaching on the Android native side cleans up cpp refs. When we detach, we need to remove the view. When it calls (for whatever reason) the attach method again, we want to make sure we're not adding the view again (thus why we don't set `isAttached` back to `false`).

All this to say - we need to explicitly removeView on detach now with the cpp cleanup logic being handled in the Android native's `onDetatchedFromWindow` as there is no guarantee the `detachFromWindow` on **this** `RiveReactNativeView` will be in sync with the native one

Reply:
@zplata This change breaks the rendering of Rive animations on our app. I guess it's because we are not using the same React Native version (0.68.5 on our side), but in our app, `onAttachedToWindow()` is called only once when we push a screen with a Rive animation and on `onDetachedFromWindow()` is called once when the screen is popped (as you would expect).

With this change, the `update()` function is no longer called because:
* `init` calls `addView(riveAnimationView)`
* `onAttachedToWindow()` checks if `childCount == 0`. It's not the case as `init` has added the view. So, `update()` is not called and the animation is not loaded

To me, the previous version relying on `onAfterUpdateTransaction()` to call the `update()` method was the proper way to do it as you might want to reload the animation if the url changes as well and not only when the animation view is attached to the window.

Reply:
Thanks @simontreny - we noticed this too.. we're looking into patching this. There's also some potential underlying `rive-android` fixes coming along that might make the coordination between the React Native view and the Android RiveNativeView a little simpler.